File: EditDistance/src/EditDistance.java
import java.util.*;

class EditDistance {

    static int compute(String str1, String str2) {
        if (str1.isEmpty()) {
            return str2.length();
        }
        if (str2.isEmpty()) {
            return str1.length();
        }

        int replace = compute(str1.substring(1), str2.substring(1)) + findNumOfReplacement(str1.charAt(0),str2.charAt(0));
        int insert = compute(str1, str2.substring(1)) + 1;

        int delete = compute(str1.substring(1), str2) + 1;
        int[] answer = new int[] {replace, insert, delete};
        return Arrays.stream(answer).min().orElse(Integer.MAX_VALUE);
    }

    static int findNumOfReplacement(char c1, char c2) {
        return c1 == c2 ? 0 : 1;
    }


    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        String s1 = scanner.next();
        String s2 = scanner.next();

        System.out.println(compute(s1, s2));
    }
}

File: edit-distance/edit_distance.py
if __name__ == '__main__':
    str1 = input()
    str2 = input()

    M = len(str1) + 1
    N = len(str2) + 1

    D = [[0]*M for _ in range(N)]
    for j in range(1, N):
        D[0][j] = D[0][j - 1] + 1

    for i in range(1, M):
        D[i][0] = D[i - 1][0] +1
        for j in range(1, N):
            if str1[i - 1] != str2[j - 1]:
                D[i][j] = min(D[i - 1][j] + 1, D[i][j - 1] + 1, D[i - 1][j - 1] + 1)
            else:
                D[i][j] = D[i - 1][j - 1]

    print(D[M - 1][N - 1])
